scripts/homebrew/template.rb

```diff
@@ -0,0 +1,63 @@
+class Iota < Formula
+    desc "Bringing the real world to Web3 with a scalable, decentralized and programmable DLT infrastructure"
+    homepage "https://www.iota.org"
+    license "Apache-2.0"
+
+    version "{{version}}"
+    checksums = {
+        "macos-arm64" => "{{macos-arm64-checksum}}",
+        "linux-x86_64" => "{{linux-x86_64-checksum}}",
+        "source" => "{{source-checksum}}",
+    }
+    @@arch = "source"
+
+    on_macos do
+        on_arm do
+            @@arch = "macos-arm64"
+        end
+    end
+
+    on_linux do
+        on_intel do
+            @@arch = "linux-x86_64"
+        end
+    end
+
+    sha256 checksums[@@arch]
+
+    if @@arch == "source"
+        depends_on "cmake" => :build
+        depends_on "libpq" => :build
+        depends_on "rust" => :build
+        on_linux do
+            depends_on "llvm" => :build
+        end
+        url "https://github.com/iotaledger/iota/archive/refs/tags/v#{version}.tar.gz"
+    else
+        url "https://github.com/iotaledger/iota/releases/download/v#{version}/iota-v#{version}-#{@@arch}.tgz"
+    end
+    
+    def install
+        if @@arch == "source"
+            ENV["GIT_REVISION"] = ""
+            system "cargo", "build", "--release", "-p", "iota", "-p", "iota-tool", "-F", "indexer,gen-completions"
```

Comment:
Should we use `--bin` instead of `-p`? Just thinking it could pull unnecessary deps and compile additional stuff for tests etc

Reply:
good call